vars/piperPipelineStageMavenStaticCodeChecks.groovy

```diff
@@ -0,0 +1,74 @@
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.GenerateStageDocumentation
+import com.sap.piper.QualityCheck
+import com.sap.piper.ReportAggregator
+
+import static com.sap.piper.Prerequisites.checkScript
+
+import groovy.transform.Field
+
+@Field String STEP_NAME = getClass().getName()
+
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field STAGE_STEP_KEYS = [
+    /** Maven modules which should be excluded by the static code checks. By default the modules 'unit-tests' and 'integration-tests' will be excluded. */
+    'mavenModulesExcludes',
+    /**Path to a filter file with bug definitions which should be excluded. */
+    'spotBugsExcludeFilterFile',
+    /**Path to a filter file with bug definitions which should be included. */
+    'spotBugsIncludeFilterFile',
+    /**A comma-separated list of exclusions (.java source files) expressed as an Ant-style pattern relative to the sources root folder, i.e. application/src/main/java for maven projects. */
+    'pmdExcludes',
+    /**The PMD rulesets to use. See the Stock Java Rulesets for a list of available rules. Defaults to a custom ruleset provided by this maven plugin. */
+    'pmdRuleSets'
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus(STAGE_STEP_KEYS)
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * Execute static code checks for Maven based projects. The plugins SpotBugs and PMD are used.
+ *
+ */
+@GenerateStageDocumentation(defaultStageName = 'staticCodeChecks')
+void call(Map parameters = [:]) {
+    String stageName = 'staticCodeChecks'//parameters.stageName?:env.STAGE_NAME
+    final script = checkScript(this, parameters) ?: null
+
+    piperStageWrapper(stageName: stageName, script: script) {
+
+        String spotBugsIncludeFilterFile = 'default_spotbugs_include_filter.xml'
+        String spotBugsLocalIncludeFilterPath = ".pipeline/${spotBugsIncludeFilterFile}"
+        writeFile file: spotBugsLocalIncludeFilterPath, text: libraryResource(spotBugsIncludeFilterFile)
+
+        String defaultPmdRulesFile = 'default_pmd_rulesets.xml'
+        String pmdRulesPath = ".pipeline/${defaultPmdRulesFile}"
+        writeFile file: pmdRulesPath, text: libraryResource(defaultPmdRulesFile)
+
+        mavenExecuteStaticCodeChecks(script: script,
+            spotBugsIncludeFilterFile: spotBugsLocalIncludeFilterPath,
+            pmdRuleSets: [pmdRulesPath])
+
+        Map configuration = ConfigurationLoader.stageConfiguration(script, stageName)
```

Comment:
Since I am currently working on a similar aspect... the config key filters you declare after the imports seem not to be used anywhere for loading and filtering the config.

Reply:
Yes but I use them for documentation purpose